RecoCTPPS/PixelLocal/interface/RPixDetTrackFinder.h

```diff
@@ -33,6 +33,7 @@ class RPixDetTrackFinder{
     void setHits(std::map<CTPPSPixelDetId, std::vector<RPixDetPatternFinder::PointInPlane> > *hitMap) {hitMap_ = hitMap; }
     virtual void findTracks()=0;
     virtual void initialize()=0;
+    virtual void addRecoInfo(int run)=0;
```

Comment:
It is not obvious, but this was done to keep this separated from the tracking algorithm.
Does it cause any issue?

Reply:
why does it need to be separate?
From your other comments it sounds like this method is needed only for a specific detector/algorithm case and will not apply to all classes derived from RPixDetTrackFinder.
The appropriate solution in this case is to either have another common base where this method is actually relevant to all downstream or really just have an embedded (not inherited/virtual) method/logic in the specific class types that need this.

Reply:
I guess the initial request of keeping the function separated came basically from the intent of keeping everything tidy and place the "hardcoded" run numbers/shift configurations outside of the tracking code, which stays cleaner. 
If you think it would be better to move this piece inside findTracks I can do it